pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/AbstractIndexType.java

```diff
@@ -62,6 +71,39 @@ public IndexReaderFactory<IR> getReaderFactory() {
     return _readerFactory;
   }
 
+  public void convertToNewFormat(TableConfig tableConfig, Schema schema) {
+    Map<String, C> deserialize = getConfig(tableConfig, schema);
+    for (Map.Entry<String, C> entry : deserialize.entrySet()) {
+      boolean fieldConfigFound = false;
+      List<FieldConfig> fieldConfigList = tableConfig.getFieldConfigList() == null
```

Comment:
Seems redundant. Can we create a map<columnName, FieldConfigList> _map and while looping over `deserialize` map, get the correct fieldConfigList for that column?